### PR TITLE
fix: Added rateLimiter to fileCreate.

### PR DESCRIPTION
**Pull request by ebadiere**
Adds HBar rate limiter to the `createFile` method

**Related issue(s)**: #2631

### CODE REVIEW

File: packages/relay/src/lib/clients/sdkClient.ts [hunk 1]
```diff
@@ -257,7 +257,7 @@ export class SDKClient {
       fileId = await this.createFile(
         ethereumTransactionData.callData,
         this.clientMain,
-        requestId,
+        requestId ?? 'create-file-from-submit-ethereum-transaction',
```
Comment (Nana-EC) [inline packages/relay/src/lib/clients/sdkClient.ts:260]: What's the goal of this?

File: packages/relay/src/lib/clients/sdkClient.ts [hunk 2]
```diff
@@ -643,60 +643,111 @@ export class SDKClient {
...
+          await this.calculateFileAppendTxTotalTinybarsCost(fileAppendTx),
+          0,
+          callerName,
+          interactingEntity,
+        );
+      }
+
+      // Ensure that the calldata file is not empty
+      if (fileId) {
+        const fileSize = await (await new FileInfoQuery().setFileId(fileId).execute(client)).size;
```
Comment (Nana-EC) [inline packages/relay/src/lib/clients/sdkClient.ts:700]: Q: does this have a cost associated with it?
If so can you open a ticket so we come back and see if this is needed still and more so also capture its cost in the metric and rate limits

File: packages/relay/src/lib/clients/sdkClient.ts [hunk 3]
```diff
@@ -643,60 +643,111 @@ export class SDKClient {
...
+            SDKClient.transactionMode,
+            fileCreateTx.constructor.name,
+            sdkClientError.status,
+            transactionFee.toTinybars().toNumber(),
+            transactionRecord?.contractFunctionResult?.gasUsed,
+            callerName,
+            interactingEntity,
+          );
+
+          this.hbarLimiter.addExpense(transactionFee.toTinybars().toNumber(), currentDateNow);
```
Comment (Nana-EC) [inline packages/relay/src/lib/clients/sdkClient.ts:731]: This only adds the file create transaction cost.
What about the fileAppend transaction(s)?

File: packages/relay/src/lib/clients/sdkClient.ts [hunk 4]
```diff
@@ -643,60 +643,111 @@ export class SDKClient {
...
+          Status.Success,
+          await this.calculateFileAppendTxTotalTinybarsCost(fileAppendTx),
+          0,
+          callerName,
+          interactingEntity,
+        );
+      }
+
+      // Ensure that the calldata file is not empty
+      if (fileId) {
```
Comment (Nana-EC) [inline packages/relay/src/lib/clients/sdkClient.ts:699]: Q: Where's the actual logic to add the costs of the FileCreate and FileAppend to the hbar rate limiter?

**Review by Nana-EC — CHANGES_REQUESTED** [review]
Let's confirm we're capturing all the FileCreate/FileAppend expenses

File: packages/relay/src/lib/clients/sdkClient.ts (hunk 1 above)
Comment (ebadiere) [reply packages/relay/src/lib/clients/sdkClient.ts:260]: The optional parameter on the requestId causes `Argument of type 'string | undefined' is not assignable to parameter of type 'string'.`. I can remove the optional all the way to the `sendRawTransaction`

Comment (Nana-EC) [reply packages/relay/src/lib/clients/sdkClient.ts:260]: My concern is the choice of string.
Doesn't seem consistent with anything we've done.
Either the requestId is provided by a user of one is generated higher up if not present.

Why do we need this workaround only here?

File: packages/relay/src/lib/clients/sdkClient.ts (hunk 3 above)
Comment (ebadiere) [reply packages/relay/src/lib/clients/sdkClient.ts:740]: Updated.

**Comment by ebadiere** [comment]
Replaced by #2634

**Pull request closed without merging**